packages/account/src/Sections/Security/ClosingAccount/closing-account-pending-conditions/closing-account-has-pending-conditions.tsx

```diff
@@ -0,0 +1,248 @@
+    onConfirm: () => void;
+};
+
+const getDerivAccount = (client_accounts: TAccounts[], login_id: string) =>
+    client_accounts.find(client_account => client_account.loginid === login_id);
+
+const getCurrentMT5Account = (mt5_login_list: DetailsOfEachMT5Loginid[], login_id: string) =>
+    mt5_login_list.find(account_obj => account_obj.login === login_id);
+
+const getCurrentDxtradeAccount = (dxtrade_accounts_list: TDetailsOfDerivXAccount[], login_id: string) =>
+    dxtrade_accounts_list.find(account_obj => account_obj.account_id === login_id);
+
+const getCurrentCtraderAccount = (ctrader_accounts_list: TDetailsOfCtraderAccount[], login_id: string) =>
+    ctrader_accounts_list.find(account_obj => account_obj.account_id === login_id);
+
+const ClosingAccountHasPendingConditions = observer(
+    ({ details, onConfirm }: TClosingAccountHasPendingConditionsProps) => {
+        const { client } = useStore();
+        const { dxtrade_accounts_list, mt5_login_list, account_list, ctrader_accounts_list } = client;
+
+        let deriv_open_positions: TDetailsOfDerivAccount[] = [];
+        let deriv_balance: TDetailsOfDerivAccount[] = [];
+        let account_pending_withdrawals: TDetailsOfDerivAccount[] = [];
+
+        let mt5_open_positions: TDetailsOfMT5Account[] = [];
+        let mt5_balance: TDetailsOfMT5Account[] = [];
+
+        let dxtrade_open_positions: TDetailsOfDerivXAccount[] = [];
+        let dxtrade_balance: TDetailsOfDerivXAccount[] = [];
+
+        let ctrader_open_positions: TDetailsOfCtraderAccount[] = [];
+        let ctrader_balance: TDetailsOfCtraderAccount[] = [];
+
+        if (details?.pending_withdrawals) {
+            Object.keys(details.pending_withdrawals).forEach(login_id => {
+                const info = {
+                    withdrawals: details.pending_withdrawals?.[login_id],
+                };
+                const deriv_account = getDerivAccount(account_list, login_id);
+                if (deriv_account) {
+                    account_pending_withdrawals = [...account_pending_withdrawals, { ...deriv_account, ...info }];
+                }
+            });
+        }
+
+        if (details?.open_positions) {
+            Object.keys(details.open_positions).forEach(login_id => {
+                const info = {
+                    positions: details.open_positions?.[login_id],
+                };
+                const deriv_account = getDerivAccount(account_list, login_id);
+                if (deriv_account) {
+                    deriv_open_positions = [...deriv_open_positions, { ...deriv_account, ...info }];
+                } else {
+                    const mt5_account = getCurrentMT5Account(mt5_login_list, login_id);
+                    if (mt5_account) {
+                        mt5_open_positions = [...mt5_open_positions, { ...mt5_account, ...info }];
+                    }
+
+                    const dxtrade_account = getCurrentDxtradeAccount(dxtrade_accounts_list, login_id);
```

Comment:
recommendation: it's always better to group all variables together, and put conditions after them all for better readability :)
>const mt5_account = getCurrentMT5Account(mt5_login_list, login_id);
>const dxtrade_account = getCurrentDxtradeAccount(dxtrade_accounts_list, login_id);
>const ctrader_account = getCurrentCtraderAccount(ctrader_accounts_list, login_id);
>
>if (mt5_account) {... 
>if (dxtrade_account) {... 
>if (ctrader_account) {...

Reply:
even though the argument names are similar, we are calling these functions from within `forEach` loops, so it's not possible to refactor as mentioned above 🙏🏼